### DISCUSSION TITLE
Special character in fulltext search throws syntax error

### DISCUSSION THREAD

**Discussion by jakecultivated** [mikro-orm/mikro-orm#4302] [Q&A]
**Describe the bug**
When trying to perform a $fulltext search against a MySQL varchar column with @ symbol in data (it is a column of email addresses), the query throws a syntax error as it seems that it is trying to parse the @ symbol as a delimiter. I'm not sure if there are any other special characters that might be causing this behavior as well. In my specific case, I was able to split the first half of the email and use that to query as a workaround, though this may not be possible in all cases. (Love Mikro-ORM by the way!)

**Stack trace**

[Nest] 7032  - 04/28/2023, 3:58:28 PM   ERROR [TempClientsService] SyntaxErrorException: select `c0`.* from `user` as `c0` left join `address` as `a1` on `c0`.`address_id` = `a1`.`id` where `c0`.`is_deleted` = false and (match(`c0`.`client_name`) against ('ClientTester' in boolean mode) or match(`a1`.`address_one`) against ('125 West Ave' in boolean mode) or match(`c0`.`email`) against ('test@mail.com' in boolean mode)) and `c0`.`role` = 'CLIENT' - syntax error, unexpected '@', expecting $end

**To Reproduce**
1. Create column with 'fulltext' index
2. Insert data in column with @ symbol
3. Attempt $fulltext query against it

**Expected behavior**
Any special characters are escaped and the query returns results as expected.

**Additional context**
Running on NestJS backend with MySQL database.

This is the query throwing the error: 
`await this.clientRepository.find(
        {
          $or: [
            { clientName: { $fulltext: tempClient.clientName } },
            {
              address: {
                addressOne: {
                  $fulltext: tempClient.address.getProperty('addressOne'),
                },
              },
            },
            { email: { $fulltext: tempClient.email } },
          ],
        },
        { populate: ['address'] },
      );`

When the email query is removed, this works perfectly.

**Versions**

| Dependency | Version |
| - | - |
| nestjs | 9.0.0 |
| typescript | 4.7.4 |
| mikro-orm | 5.5.3 |
| mikro-orm/mysql | 5.5.3 |

**Comment by B4nan** [comment]
You will need to wrap it inside double quotes:

```ts
em.find(Client, { email: { $fulltext: `"${tempClient.email}"' } });
```

https://stackoverflow.com/questions/57381338/mysql-syntax-error-unexpected-expecting-end
